File: March_2024/D3/Solution.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        pre_head = ListNode(0,head)
        front = back = pre_head
        while front and n >= 0:
            front = front.next
            n -= 1
        while front:
            front = front.next
            back = back.next
        back.next = back.next.next
        return pre_head.next

        

            

File: March_2024/D3/Solution.java
package March_2024.D3;

//Definition for singly-linked list.
class ListNode {
    int val;
    ListNode next;

    ListNode() {
    }

    ListNode(int val) {
        this.val = val;
    }

    ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }
}

public class Solution {
    public static void main(String[] args) {

    }

    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode front, rear;
        front = head;
        rear = head;
        for (int i = 0; i < n; i++) {
            if (front.next == null) {
                return head.next;
            }
            front = front.next;
        }
        while (front.next != null) {
            rear = rear.next;
            front = front.next;
        }
        rear.next = rear.next.next;
        return head;
    }
}
